Sources/StreamChatUI/Utils/UIImageView+Extensions.swift

```diff
@@ -6,9 +6,34 @@ import Nuke
 import UIKit
 
 extension UIImageView {
-    func loadImage(from url: URL?, placeholder: UIImage? = nil) {
+    func loadImage(from url: URL?, placeholder: UIImage? = nil, resizeAutomatically: Bool = true) {
+        guard !SystemEnvironment.isTests else {
+            // When running tests, we load the images synchronously
+            image = url.flatMap { UIImage(data: try! Data(contentsOf: $0)) }
+            return
+        }
+
+        // Cancel any previous loading task
+        currentImageLoadingTask?.cancel()
+
         guard let url = url else { image = nil; return }
+
+        let preprocessors: [ImageProcessing] = resizeAutomatically && bounds.size != .zero
+            ? [ImageProcessors.Resize(size: bounds.size, contentMode: .aspectFill, crop: true)]
+            : []
+ 
+        let request = ImageRequest(url: url, processors: preprocessors)
         let options = ImageLoadingOptions(placeholder: placeholder)
-        Nuke.loadImage(with: url, options: options, into: self)
+
+        currentImageLoadingTask = Nuke.loadImage(with: request, options: options, into: self)
+    }
+}
+
+private extension UIImageView {
+    static var nukeLoadingTaskKey: UInt8 = 0
+
+    var currentImageLoadingTask: ImageTask? {
+        get { objc_getAssociatedObject(self, &Self.nukeLoadingTaskKey) as? ImageTask }
+        set { objc_setAssociatedObject(self, &Self.nukeLoadingTaskKey, newValue, .OBJC_ASSOCIATION_RETAIN) }
```

Comment:
Why do we need this? 🤔

Reply:
To be able to cancel previous tasks. It's not related to resizing. If we don't have this the following can happen:
```
loadImage A -----X-----------------------------------------------------------X
            start loading A                                              image = loaded Image A

cell is reused

loadImage B -----------------------X--------------------------------X----------
                       start loading B            image = loaded Image B
```
If we don't cancel the loading task for A explicitly, what can happen is that it will finish after the loading for another task has finished.

Reply:
Ah nevermind, I see, this is an extension of UIImageView, so it is a way to add a new property to UIImageView, without making a subclass right?

Reply:
Exactly, we don't want to limit the users to use our specific `UIImageView` subclass for this.

Reply:
Yes my question was not about why we needed the cancel, that I understood, my questions was why we needed the swizzling 👍